data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwarderAppConfig.java

```diff
@@ -96,10 +96,10 @@ public Server server(
     }
 
     @Bean
-    public PeerForwarderServer peerForwarderServer(
+    public PeerForwarderServerProxy peerForwarderServerProxy(
```

Comment:
You can keep the original signature here. I think it is a good signal that this is the `peerForwarderServer` which should be used. The user of the bean doesn't need to care if it is the proxy or not. That is an internal detail.

```
public PeerForwarderServer peerForwarderServer(...)
```

Reply:
`PeerForwarderServer` is an interface now, I guess we've to create the `PeerForwarderServerProxy` bean here.

Reply:
From the consumer of the bean, you are creating a `PeerForwarderServer`. You still return a `PeerForwarderServerProxy`, but it is provided as the interface.

This should work:

```
public PeerForwarderServer peerForwarderServer(
     @Qualifier("peerForwarderServer") final Server peerForwarderServer,
     final PeerForwarderConfiguration peerForwarderConfiguration) {
 return new PeerForwarderServer(peerForwarderConfiguration, peerForwarderServer);
 return new PeerForwarderServerProxy(peerForwarderConfiguration, peerForwarderServer);
}
```